packages/plugin-core/src/commands/Doctor.ts

```diff
@@ -288,6 +288,14 @@ export class DoctorCommand extends BasicCommand<CommandOpts, CommandOutput> {
     return { installStatus, contents };
   }
 
+  async reload(opts: CommandOpts) {
```

Comment:
can we avoid casting as per [[Going outside of the typescript compiler|dendron://dendron.docs/dev.process.code.best-practices#going-outside-of-the-typescript-compiler]]?

in this case, I don't think we're doing anything with the output so there's no need to cast.

also, can we add a docstring here?

Reply:
can we also make it private? 

I see that it's exposed so that we can do a `sinon.spy(cmd, "reload");` in the test - my take (others may disagree), it's not really worth having that test to verify this functionality that's internal to this class.

Reply:
i think its worhtwile because this is what this case is fixing (that we don't reload for this check). we can use [[Exposing private methods to tests|dendron://dendron.docs/dev.process.qa.test#exposing-private-methods-to-tests]] to expose it for testing

Reply:
We could also keep it private, then do a cast when using sinon. Since public/private is resolved at type level only, the spy will still work.
```ts
const reloadSpy = sinon.spy(cmd, "reload" as keyof DoctorCommand);
```

Reply:
Casting was done because `ReloadIndexCommand` can return undefined, and all places that needed it down the line required `engine` to not be undefined. 
(and also didn't want to duplicate code by type checking on every single switch case, but that was me being a little lazy 😄)


I changed it so that `this.reload` checks for undefined before it returns, and throws if it is. 

Also just called reload on each switch case where it needed it. (I think typescript is not smart enough to narrow `DEngineClient | undefined` to `DEngineClient` based on the switch statement it lands on)

> We could also keep it private, then do a cast when using sinon. Since public/private is resolved at type level only, the spy will still work.

Nice tip! never thought of this.